### PR TITLE
Fix Sontaype pom validator

### LINKED ISSUE #64: Fix maven publish: POM validation
Sonatype validation failed upon "closing" the staged repo:

typeId | pom-staging
-- | --
failureMessage | Invalid  POM:  /io/honeycomb/honeycomb-opentelemetry-javaagent/0.2.0/honeycomb-opentelemetry-javaagent-0.2.0.pom:  Project description missing
failureMessage | Invalid POM: /io/honeycomb/honeycomb-opentelemetry-sdk/0.2.0/honeycomb-opentelemetry-sdk-0.2.0.pom: Project description missing
failureMessage | Invalid  POM:  /io/honeycomb/honeycomb-opentelemetry-common/0.2.0/honeycomb-opentelemetry-common-0.2.0.pom:  Project description missing

### PR DESCRIPTION
**Pull request by vreynolds**
* add description to POMs
* add note about testing staged artifacts
* Fixes #64 

Tested locally: publish to Sonatype, close (validation passed), use in test app.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -64,6 +64,7 @@ subprojects {
                     }
                     pom {
                         name = 'Honeycomb OpenTelemetry Distribution for Java'
+                        description = 'Honeycomb distribution of the OpenTelemetry Java Instrumentation.'
```
Comment (robbkidd) [inline build.gradle:67]: Certainly it's simpler to have this one description applied to all the sub-projects for this go around. Is it possible to set the POM description individually on sub-projects in the future to describe their differences? For example:

* honeycomb-opentelemetry-javaagent: "The JavaAgent provides auto-instrumentation and a configurable default tracer."
* honeycomb-opentelemetry-sdk: "The SDK for manual instrumentation without the JavaAgent."
* honeycomb-opentelemetry-common: "Functionality shared across components of the Honeycomb OpenTelemetry distribution."

**Review by robbkidd — APPROVED** [review]
👍🏻 

... but with a question about whether distinct POM descriptions for the subprojects is possible in future iterations.

**Comment by vreynolds** [comment]
@robbkidd added the distinct descriptions, keeping them a bit more generic, based on "conventions" I saw in maven

**Review by robbkidd — APPROVED** [review]
Yes!

**Pull request merged**